jsonutil/puritan/puritan_test.go

```diff
@@ -0,0 +1,239 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package puritan_test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/jsonutil/puritan"
+)
+
+func Test(t *testing.T) { check.TestingT(t) }
+
+type escapeSuite struct{}
+
+var _ = check.Suite(escapeSuite{})
+
+var table = map[string]string{
+	"null":           "",
+	`"hello"`:        "hello",
+	`"árbol"`:        "árbol",
+	`"\u0020"`:       " ",
+	`"\uD83D\uDE00"`: "😀",
```

Comment:
Is that right? Do we want emojis in these fields?

Reply:
do you want to tell our devs they _can't_ use emoji in descriptions?

Reply:
I want for it to at least be a conscious decision and not a side-effect of not talking about it.

Reply:
for the record, I think we _do_ want to allow emojis in descriptions. Disallowing it feels overly prescriptive of us.

There are bound to be things using emojis as part of their name, for example.